src/jaxsim/high_level/model.py

```diff
@@ -1414,20 +1414,10 @@ def integrate(
             physics_model=self.data.model_input
         )
 
-        if integrator_type is IntegratorType.EulerForward:
-            integrator_fn = ode_integration.ode_integration_euler
-
-        elif integrator_type is IntegratorType.EulerSemiImplicit:
-            integrator_fn = ode_integration.ode_integration_euler_semi_implicit
-
-        elif integrator_type is IntegratorType.RungeKutta4:
-            integrator_fn = ode_integration.ode_integration_rk4
-
-        else:
-            raise ValueError(integrator_type)
+        assert isinstance(integrator_type, IntegratorType)
 
         # Integrate the model dynamics
-        ode_states, aux = integrator_fn(
+        ode_states, aux = ode_integration._integrator_registry[integrator_type](
```

Comment:
@flferretti I missed this in the review, I suspect you have to call the new generic `ode_fixed_step`, maybe creating a lambda.

Reply:
Yes, I was checking it right now. I'm going to do some tests and get back to you asap

Reply:
Fixed in https://github.com/ami-iit/jaxsim/pull/78/commits/73f8ca6d53f526730f07ded77ec2b521c3427390